cli/utils.go

```diff
@@ -0,0 +1,151 @@
+package cli
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"sort"
+	"strings"
+
+	"github.com/spf13/viper"
+
+	"github.com/prometheus/alertmanager/pkg/parse"
+	"github.com/prometheus/alertmanager/types"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/pkg/labels"
+)
+
+type ByAlphabetical []labels.Matcher
+
+func (s ByAlphabetical) Len() int      { return len(s) }
+func (s ByAlphabetical) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
+func (s ByAlphabetical) Less(i, j int) bool {
+	if s[i].Name != s[j].Name {
+		return s[i].Name < s[j].Name
+	} else if s[i].Type != s[j].Type {
+		return s[i].Type < s[j].Type
+	} else if s[i].Value != s[j].Value {
+		return s[i].Value < s[j].Value
+	} else {
+		return false
+	}
+}
+
+func GetAlertmanagerURL() (*url.URL, error) {
+	u, err := url.ParseRequestURI(viper.GetString("alertmanager.url"))
+	if err != nil {
+		return nil, errors.New("Invalid alertmanager url")
+	}
+	return u, nil
+}
+
+// Parse a list of labels (cli arguments)
+func parseMatchers(inputLabels []string) ([]labels.Matcher, error) {
+	matchers := make([]labels.Matcher, 0)
+
+	for _, v := range inputLabels {
+		fmt.Println(v)
```

Comment:
Sorry debugging line. Was just trying to get it into a state where I could show the issues.